Products/ZenEvents/zentrap.py

```diff
@@ -577,9 +600,15 @@ def _process_varbinds(self, varbinds):
             # Record the varbind instance(s) in their 'raw' form.
             for full_name, value in data:
                 suffix = full_name[offset:]
-                result[full_name].append(value)
-                if suffix:
-                    result[base_name + ".sequence"].append(suffix)
+
+                if self.varbindCopyMode == LEGACY_VARBIND_COPY_MODE:
+                    result[base_name].append(value)
+                    if suffix:
+                        result[base_name + ".ifIndex"].append(suffix)
+                else:
+                    result[full_name].append(value)
+                    if suffix:
+                        result[base_name + ".sequence"].append(suffix)          
 
         return {name: ','.join(vals) for name, vals in result.iteritems()}
```

Comment:
Rather than mixing the various copy modes into the same method, it'll be more clear to future readers of this code if the different copy modes were implemented as distinct functions.  To avoid testing the `varbindCopyMode` variable multiple times, each copy mode deserves its own class which is selected and initialized when zentrap starts.

Here's what the "legacy" class would look like:
```
class _LegacyVarbindProcessor(object):

    def __init__(self, oid2name):
        self.oid2name = oid2name

    def __call__(self, varbinds):
        result = defaultdict(list)
        for oid, value in varbinds:
            base_name = self.oid2name(oid, exactMatch=False, strip=True)
            full_name = self.oid2name(oid, exactMatch=False, strip=False)
            if base_name != full_name:
                result[base_name].append(value)
                suffix = full_name[len(base_name) + 1:]
                result[base_name + ".ifIndex"].append(suffix)
            else:
                result[full_name].append(value)
        return result
```
This is what the "direct" mode class would look like:
```
class _DirectVarbindProcessor(object):

    def __init__(self, oid2name):
        self.oid2name = oid2name

    def __call__(self, varbinds):
        result = defaultdict(list)
        for oid, value in varbinds:
            base_name = self.oid2name(oid, exactMatch=False, strip=True)
            full_name = self.oid2name(oid, exactMatch=False, strip=False)
            result[full_name].append(value)
            if base_name != full_name:
                suffix = full_name[len(base_name) + 1:]
                result[base_name + ".sequence"].append(suffix)
        return result
```
The "mixed" mode class will be similar and its `__call__` method I'll leave as an exercise for you.
```
class _MixedVarbindProcessor(object):

    def __init__(self, oid2name):
        self.oid2name = oid2name

    def __call__(self, varbinds):
        ...
```
To select the appropriate varbind processor, I suggest the following:
```
# defined as a module or TrapTask class variable
_varbind_processors = {
    LEGACY_VARBIND_COPY_MODE: _LegacyVarbindProcessor.
    DIRECT_VARBIND_COPY_MODE: _DirectVarbindProcessor,
    MIXED_VARBIND_COPY_MODE: _MixedVarbindProcessor,
}
```
Then, somewhere in the `TrapTask`'s `__init__` method:
```
    processor_class = _varbind_processors.get(self.varbindCopyMode)
    self._process_varbinds = processor_class(self.oid2name)
```